packages/curve-ui-kit/src/lib/model/query/factory.ts

```diff
@@ -75,7 +65,7 @@ export function queryFactory<
     prefetchQuery: (params, staleTime = 0) =>
       queryClient.prefetchQuery({ queryKey: queryKey(params), queryFn, staleTime }),
     fetchQuery: (params, options) => queryClient.fetchQuery({ ...getQueryOptions(params), ...options }),
-    useQuery: createQueryHook(getQueryOptions),
+    useQuery: (params, condition) => useQuery(getQueryOptions(params, condition)),
```

Comment:
Excuse me for adding a commit to your PR, but imo TanStack shines when you inline stuff. The moment you are creating wrappers you're stuck in type-and-generic hell. We can get rid of `createQueryHook` in its entirety, together with all of the complex typing, by just directly inline here. This was literally the only usage of that function anyway.

Reply:
Good catch, thanks, that did not make any sense anymore after this PR indeed